### PR TITLE
Changes to VATemplate, SkillTemplate and skills to use the new skills lib

### PR DESCRIPTION
**Pull request by lzc850612**
## Description

Remove ContinueDialogAsync call on skill because we don't require it to pass the activity to skill.

DO NOT MERGE this PR because we have to wait until the new version of Microsoft.Bot.Builder.Skills to use this.

## Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the appropriate unit tests
- [ ] I have tested any new/updated dialogs using Speech in the emulator to ensure the [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) property is set to enable a high quality speech-first experience and the appropriate [InputHints](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) are set correctly. 
- [ ] I have updated related documentation

- [ ] A duplicate issue is filed to track future  work.

- [ ] All languages have been updated
- [ ] You have tested deployment with your new models

### CODE REVIEW

File: templates/Virtual-Assistant-Template/csharp/Sample/VirtualAssistantSample/Dialogs/MainDialog.cs [hunk 1]
```diff
@@ -22,373 +22,374 @@
...
+                }
+            }
+
+            var forward = true;
+            var ev = dc.Context.Activity.AsEventActivity();
+            if (!string.IsNullOrWhiteSpace(ev.Name))
+            {
+                switch (ev.Name)
+                {
+                    case Events.TimezoneEvent:
```
Comment (lzc850612) [inline templates/Virtual-Assistant-Template/csharp/Sample/VirtualAssistantSample/Dialogs/MainDialog.cs:208]: these are the main changes. others are all indentation.

File: skills/src/csharp/pointofinterestskill/pointofinterestskill/Dialogs/MainDialog.cs [hunk 2]
```diff
@@ -275,39 +275,38 @@ protected override async Task<InterruptionAction> OnInterruptDialogAsync(DialogC
...
-				}
-			}
-		}
-
-		private async Task<InterruptionAction> OnCancel(DialogContext dc)
+        }
+
+        private async Task PopulateStateFromSkillContext(ITurnContext context)
+        {
+            // If we have a SkillContext object populated from the SkillMiddleware we can retrieve requests slot (parameter) data
```
Comment (darrenj) [inline skills/src/csharp/pointofinterestskill/pointofinterestskill/Dialogs/MainDialog.cs]: The SkillContext concept has gone away on the Skill side, perhaps we should rename this method?

**Review by darrenj — CHANGES_REQUESTED** [review]

**Comment by ryanisgrig** [comment]
FYI @lzc850612 in the future you can create a Draft PR that will show the changes but not allow it to be merged until you switch it until ready for review

File: skills/src/csharp/pointofinterestskill/pointofinterestskill/Dialogs/MainDialog.cs (hunk 2 above)
Comment (lzc850612) [reply skills/src/csharp/pointofinterestskill/pointofinterestskill/Dialogs/MainDialog.cs]: good catch

**Comment by lzc850612** [comment]
> FYI @lzc850612 in the future you can create a Draft PR that will show the changes but not allow it to be merged until you switch it until ready for review

yeah i forgot to do this. good tip

**Review by lauren-mills — APPROVED** [review]

**Pull request merged**